src/app/contactmanager/components/main-content/main-content.component.ts

```diff
@@ -32,14 +34,15 @@ export class MainContentComponent implements OnInit {
   
   private route = inject(ActivatedRoute);
   private userService = inject(UserService);
+  users$ = toObservable(this.userService.users);
 
   ngOnInit(): void {
     this.route.paramMap.subscribe((params: ParamMap) => {
       let id = Number(params.get('id'));
       if (id == 0) id = 1;      
       this.user = null;
       
-      this.userService.users.subscribe(users => {
+      this.users$.subscribe(users => {
```

Comment:
This is pseudo code so you'll want to check that it works, but something like this will clean up your component and use some of the latest features that are in v17.2

Signal inputs are technically in dev preview, so if you wanted to avoid those you can still get the route params through an input in which case you can leverage a setter to use their value without having to inject the activated route and subscribe to it.
```
export class MainContentComponent implements OnInit {
  private userService = inject(UserService);

  // Signal inputs are in dev preview in 17.2
  // componentInputBinding has to be enabled where your router is initialized - app.config.ts
  // this fetches the id from the query params
  id = input<string>;

  // computed here because we are using two signals to get a new value
  selectedUser = computed(() => this.userService.userById(this.id()));
  
  // we can pipe a delay off of the observable returned by toObservable
  // this gets rid of the need to use setTimeout in a nested subscription in OnInit
  // subscribe to the observable in the template using the async pipe
  users$ = toObservable(this.selectedUser).pipe(delay(500));
}

Reply:
This is the only part, where I failed to follow your advice.

I did the input part, but didn't get the computed(() => ..) to work, it simply didn't fire, did I miss something fundamental? 

The pipe/delay was really nice as well, component cleaned up pretty nice even if I didn't understand the computed-part.